internal/plugin/connectors/http/generic/connector.go

```diff
@@ -39,11 +41,22 @@ func (c *Connector) Connect(
 	}
 
 	// Add configured params to request
-	params, err := renderTemplates(c.config.QueryParams, credentialsByID)
+	queryParams, err := renderTemplates(c.config.QueryParams, credentialsByID)
 	if err != nil {
 		return fmt.Errorf("failed to render query params: %s", err)
 	}
-	r.URL.RawQuery = appendQueryParams(*r.URL, params)
+	r.URL.RawQuery = appendQueryParams(*r.URL, queryParams)
+
+	// Add OAuth1 Authorization Header to the request
+	oauth1params, err := renderTemplates(c.config.OAuth1, credentialsByID)
```

Comment:
```suggestion
	oauth1Params, err := renderTemplates(c.config.OAuth1, credentialsByID)
```

Reply:
This whole block should probably only happen if oauth info is specified but we run it every time

Reply:
We need to run `renderTemplates()` to check for the `oauth1` section in config, like `queryParams` does. This looks for `oauth1` and returns an `err` if it isn't there. I agree with moving the if statement after that to only be used if `renderTemplates()` doesn't return an `err`. But how would you suggest checking for this instead?